error-handler.js

```diff
@@ -98,15 +99,16 @@ var mixIn = require('mout/object/mixIn'),
     stream.pipe(res);
   },
 
-  send = function send(statusCode, err, res, o) {
+  send = function send(statusCode, err, res, o) {    
+    if (!err.message) err.message = statusCodes[statusCode];
```

Comment:
That works, but I like create object use better as it avoids whitelist.

On Wed, Mar 9, 2016, 11:53 PM Eric Elliott <notifications@github.com> wrote:

> In error-handler.js
> <https://github.com/ericelliott/express-error-handler/pull/12#discussion_r55633932>
> :
>
> > @@ -98,15 +99,16 @@ var mixIn = require('mout/object/mixIn'),
> >      stream.pipe(res);
> >    },
> >
> > -  send = function send(statusCode, err, res, o) {
> > +  send = function send(statusCode, err, res, o) {
> > +    if (!err.message) err.message = statusCodes[statusCode];
>
> How about this?
>
> send = function send(statusCode, err, res, o) {
>   var base = {
>     status: statusCode
>   };
>   var extended = {};
>
>   if (!err.message) base.message = statusCodes[statusCode];
>   if (err.name) extended.name = err.name;
>
>   var body = (o.serializer) ?
>     o.serializer(mixIn({}, err, base, extended)) :
>     base;
>
>   res.status(statusCode);
>   res.send(body);
> }
>
> IMO, hardcoding a key whitelist is more responsible than blindly mutating
> the err object, particularly since we're just talking about what is being
> sent to the user. The full error can be inspected and logged elsewhere by
> other middleware for other purposes.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ericelliott/express-error-handler/pull/12/files#r55633932>
> .
>